app/v2/notifications/post_notifications.py

```diff
@@ -312,84 +312,6 @@ def process_document_uploads(
     return personalisation_data
 
 
-def process_letter_notification(
-    *,
-    letter_data,
-    api_key,
-    template,
-    reply_to_text,
-    precompiled=False,
-):
-    if api_key.key_type == KEY_TYPE_TEAM:
-        raise BadRequestError(message='Cannot send letters with a team api key', status_code=403)
-
-    if not api_key.service.research_mode and api_key.service.restricted and api_key.key_type != KEY_TYPE_TEST:
-        raise BadRequestError(message='Cannot send letters when service is in trial mode', status_code=403)
-
-    if precompiled:
-        return process_precompiled_letter_notifications(
-            letter_data=letter_data, api_key=api_key, template=template, reply_to_text=reply_to_text
-        )
-
-    test_key = api_key.key_type == KEY_TYPE_TEST
-
-    # if we don't want to actually send the letter, then start it off in SENDING so we don't pick it up
-    status = NOTIFICATION_CREATED if not test_key else NOTIFICATION_SENDING
-    queue = QueueNames.CREATE_LETTERS_PDF if not test_key else QueueNames.RESEARCH_MODE
-
-    notification = create_letter_notification(
```

Comment:
I believe app.notifications.process_letter_notifications.py can now be deleted because this is the only place `create_letter_notification` is used.

Reply:
I've removed this code.